### PR TITLE
[f] Move Asset into App Storage

### PR DESCRIPTION
**Pull request by thuyenBitmark**

### CODE REVIEW

File: BitmarkRegistry/Helpers/BitmarkFileUtil.swift [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+//
+
+import Foundation
+
+class BitmarkFileUtil {
+  static func moveFile(from fileURL: URL, toFolder downloadedFolderURL: URL) throws {
+    let filename = fileURL.lastPathComponent
+    let destinationURL = downloadedFolderURL.appendingPathComponent(filename)
+    let data = try Data(contentsOf: fileURL)
+    try data.write(to: destinationURL, options: [.completeFileProtection, .atomic])
```
Comment (anhnguyenbitmark) [inline BitmarkRegistry/Helpers/BitmarkFileUtil.swift]: Can we use [this](https://developer.apple.com/documentation/foundation/filemanager/1414750-moveitem) to do moving item? Reading file data to app's memory is not a good idea.

File: BitmarkRegistry/Services/AssetFileService.swift [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+//
+
+import Foundation
+import BitmarkSDK
+
+class AssetFileService {
+
+  // MARK: Properties
+  var owner: Account
+  var assetId: String
```
Comment (anhnguyenbitmark) [inline BitmarkRegistry/Services/AssetFileService.swift]: I don't see we need to modify this value anywhere outside `init` method. Why do we need `var` here?

Comment (thuyenBitmark) [reply BitmarkRegistry/Services/AssetFileService.swift]: my bad; thanks for pointing it out :)

File: BitmarkRegistry/Helpers/BitmarkFileUtil.swift [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+//  FileUtil.swift
+//  BitmarkRegistry
+//
+//  Created by Macintosh on 6/28/19.
+//  Copyright © 2019 thuyentruong. All rights reserved.
+//
+
+import Foundation
+
+class BitmarkFileUtil {
```
Comment (anhnguyenbitmark) [inline BitmarkRegistry/Helpers/BitmarkFileUtil.swift]: Empty class.

File: BitmarkRegistry/Services/AssetFileService.swift [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  // MARK: Properties
+  let owner: Account
+  let assetId: String
+
+  lazy var downloadedFolderURL: URL = {
+    let directoryURL = URL(
+      fileURLWithPath: owner.getAccountNumber() + "/assets/" + assetId + "/downloaded",
+      relativeTo: FileManager.sharedDirectoryURL
+    )
+    try? FileManager.default.createDirectory(at: directoryURL, withIntermediateDirectories: true)
```
Comment (anhnguyenbitmark) [inline BitmarkRegistry/Services/AssetFileService.swift]: We need to add TODO here to handle error.
Later on I'll add Sentry for error reporting.

File: BitmarkRegistry/Helpers/BitmarkFileUtil.swift (hunk 3 above)
Comment (thuyenBitmark) [reply BitmarkRegistry/Helpers/BitmarkFileUtil.swift]: I'm aware it; but then I'm implementing the encrypted function in this File in the next PR @@

File: BitmarkRegistry/Services/AssetFileService.swift (hunk 4 above)
Comment (thuyenBitmark) [reply BitmarkRegistry/Services/AssetFileService.swift]: `try?` is on purpose (following the tutorial video)
this code line will cover: create directory if the directory has not existed yet; if not, nothing happens

Comment (anhnguyenbitmark) [reply BitmarkRegistry/Services/AssetFileService.swift]: Ok.

**Pull request merged**